### PR TITLE
TablePanel: Fix vertical scrollbar

### LINKED ISSUE #52136: Table Scrollbars not always usable
**What happened**:

When using Grafana Explore and results are in table format, vertical and horizontal scrollbars of the table may be hard to use because they are outside the visible area (see below).

Depending on browser / OS settings (?), the barely visible scrollbars may even fade out and become completely invisible, resulting in a frustrating experience (See video especially).

May also be an UX design issue / Accessibility issue?

**What you expected to happen**:

Table scrollbars should always be accessible.

**How to reproduce it (as minimally and precisely as possible)**:

- Get a result table with some data in Explore, e.g. use this link: https://play.grafana.org/explore?orgId=1&left=%7B%22datasource%22:%22Prometheus:%20Robust%20Perception%20Sample%20Endpoint%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22topk(100,%20%7Bjob%3D~%5C%22.%2B%5C%22%7D)%22,%22range%22:false,%22editorMode%22:%22code%22,%22instant%22:true,%22exemplar%22:false%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D
- The table has several columns and rows, requiring scrollbars in both directions

### Issue 1) Vertical Scrollbar not accessible

The table has to be scrolled to the far right to get to the vertical scrollbar

grafik

GIF:

explorer_jKmcWk1haH

### Issue 2) Horizontal Scrollbar not accessible

The table does seem to have a preferred height, which may lead to the horizontal scrollbar also not being accessible.
The whole explore pages gets a vertical scrollbar, which is barely visible at all.

Using the mouse wheel may not always help, because if hovering on the data, the table gets scrolled, not the explore page.

OK if enough vertical space:

grafik

After expanding panels NOK:

grafik

Video:

https://user-images.githubusercontent.com/33002073/178688784-3a8a58f1-d2a2-47f2-a238-83fe4efb50a2.mp4

**Anything else we need to know?**:

Screenshots are from browsers running at 1920x1080, 100% zoom.

Not sure if this issue also applies to dashboards or only explore.

**Environment**:
- Grafana version: `8.5.0` and `9.0.3`
- User OS & Browser: Windows, Firefox `102.0.1` and Google Chrome `103`

### PR DESCRIPTION
**Pull request by mdvictor**
**What this PR does / why we need it**:
Vertical scroll bar in table panel did not use the custom scrollbar component (it defaulted to browser scrollbar) and thus functionality was limited and not on-par with other scrollbars in Grafana.

Tested on Chrome, Safari and Firefox.

Screenshot 2022-08-09 at 14 17 44

Screenshot 2022-08-09 at 14 45 15

**Which issue(s) this PR fixes**:

Fixes #52136 

**Special notes for your reviewer**:

Here is an example JsonPanel with different tables (different heights, data, pagination settings) for testing:

<details open>
<pre>
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "grafana",
          "uid": "-- Grafana --"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "target": {
          "limit": 100,
          "matchAny": false,
          "tags": [],
          "type": "dashboard"
        },
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 552,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "datasource": {
        "type": "testdata",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "align": "auto",
            "displayMode": "auto",
            "inspect": false
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 9,
        "w": 12,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "footer": {
          "fields": "",
          "reducer": [
            "sum"
          ],
          "show": false
        },
        "showHeader": true
      },
      "pluginVersion": "9.2.0-pre",
      "targets": [
        {
          "csvContent": "test1,test2,test3,test4,test5,test6,test7,test8,test9\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"",
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "refId": "A",
          "scenarioId": "csv_content"
        }
      ],
      "title": "Panel Title",
      "type": "table"
    },
    {
      "datasource": {
        "type": "testdata",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "align": "auto",
            "displayMode": "auto",
            "inspect": false
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 10,
        "w": 11,
        "x": 12,
        "y": 0
      },
      "id": 6,
      "options": {
        "footer": {
          "enablePagination": true,
          "fields": "",
          "reducer": [
            "sum"
          ],
          "show": false
        },
        "showHeader": true
      },
      "pluginVersion": "9.2.0-pre",
      "targets": [
        {
          "csvContent": "test1,test2,test3,test4,test5,test6,test7,test8,test9\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"",
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "refId": "A",
          "scenarioId": "csv_content"
        }
      ],
      "title": "Panel Title",
      "type": "table"
    },
    {
      "datasource": {
        "type": "testdata",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "align": "auto",
            "displayMode": "auto",
            "inspect": false
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 9,
        "w": 12,
        "x": 0,
        "y": 9
      },
      "id": 3,
      "options": {
        "footer": {
          "fields": "",
          "reducer": [
            "sum"
          ],
          "show": false
        },
        "showHeader": true
      },
      "pluginVersion": "9.2.0-pre",
      "targets": [
        {
          "csvContent": "test1,test2,test3,test4\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"",
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "refId": "A",
          "scenarioId": "csv_content"
        }
      ],
      "title": "Panel Title",
      "type": "table"
    },
    {
      "datasource": {
        "type": "testdata",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "align": "auto",
            "displayMode": "auto",
            "inspect": false
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 18,
        "w": 11,
        "x": 12,
        "y": 10
      },
      "id": 7,
      "options": {
        "footer": {
          "enablePagination": true,
          "fields": "",
          "reducer": [
            "sum"
          ],
          "show": false
        },
        "showHeader": true
      },
      "pluginVersion": "9.2.0-pre",
      "targets": [
        {
          "csvContent": "test1,test2,test3,test4,test5,test6,test7,test8,test9\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data5\",\"data6\",\"data7\",\"data8\",\"data9\"",
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "refId": "A",
          "scenarioId": "csv_content"
        }
      ],
      "title": "Panel Title",
      "type": "table"
    },
    {
      "datasource": {
        "type": "testdata",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "align": "auto",
            "displayMode": "auto",
            "inspect": false
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 10,
        "w": 10,
        "x": 0,
        "y": 18
      },
      "id": 4,
      "options": {
        "footer": {
          "fields": "",
          "reducer": [
            "sum"
          ],
          "show": false
        },
        "showHeader": true
      },
      "pluginVersion": "9.2.0-pre",
      "targets": [
        {
          "csvContent": "test1,test2,test3,test4\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\"",
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "refId": "A",
          "scenarioId": "csv_content"
        }
      ],
      "title": "Panel Title",
      "type": "table"
    },
    {
      "datasource": {
        "type": "testdata",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "align": "auto",
            "displayMode": "auto",
            "inspect": false
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 6,
        "w": 16,
        "x": 0,
        "y": 28
      },
      "id": 5,
      "options": {
        "footer": {
          "fields": "",
          "reducer": [
            "sum"
          ],
          "show": false
        },
        "showHeader": true
      },
      "pluginVersion": "9.2.0-pre",
      "targets": [
        {
          "csvContent": "test1,test2,test3,test4,test5,test6,test7\n\"data1\",\"data2\",\"data3\",\"data4\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data2\",\"data3\",\"data4\"\n\"data1\",\"data2\",\"data3\",\"data4\",\"data2\",\"data3\",\"data4\"",
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "refId": "A",
          "scenarioId": "csv_content"
        }
      ],
      "title": "Panel Title",
      "type": "table"
    }
  ],
  "schemaVersion": 37,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {},
  "timezone": "",
  "title": "tableScrollbarsIssue",
  "uid": "PGYg5Wm4k",
  "version": 7,
  "weekStart": ""
}
</pre>
</details>

### CODE REVIEW

**Review by torkelo — CHANGES_REQUESTED** [review]
Sadly this breaks virtualized rendering (Provided by FixedSizeList), this is a very important performance requirement when rendering many rows. Testing this PR disables this and loading a test dashboard like  http://localhost:3000/d/U_bZIMRMk/panel-tests-react-table?orgId=1  (https://github.com/grafana/grafana/blob/main/devenv/dev-dashboards/panel-table/table_tests_new.json)  results in super slow renderings of some of the tables in that dashboard. 

To setup the devenv dashboards run setup.sh in devenv folder

**Comment by mdvictor** [comment]
> Sadly this breaks virtualized rendering (Provided by FixedSizeList), this is a very important performance requirement when rendering many rows. Testing this PR disables this and loading a test dashboard like http://localhost:3000/d/U_bZIMRMk/panel-tests-react-table?orgId=1 (https://github.com/grafana/grafana/blob/main/devenv/dev-dashboards/panel-table/table_tests_new.json) results in super slow renderings of some of the tables in that dashboard.
> 
> To setup the devenv dashboards run setup.sh in devenv folder

I see, there is definitely worse performance with this. I'll look more into it the virtual rendering thing.

File: packages/grafana-ui/src/components/CustomScrollbar/CustomScrollbar.tsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ interface Props {
   setScrollTop?: (position: ScrollbarPosition) => void;
   autoHeightMin?: number | string;
   updateAfterMountMs?: number;
+  onCustomScroll?: React.UIEventHandler;
```
Comment (dprokop) [inline packages/grafana-ui/src/components/CustomScrollbar/CustomScrollbar.tsx:26]:
```suggestion
  onScroll?: React.UIEventHandler;
```

**Review by dprokop — APPROVED** [review]
As far as I see, the current solution is actually working with the virtualized list!

**Review by torkelo — APPROVED** [review]

**Pull request merged**